packages/fxa-content-server/app/scripts/views/mixins/modal-panel-mixin.js

```diff
@@ -33,8 +36,66 @@ export default {
 
     this.$el.on($.modal.AFTER_CLOSE, () => this.onAfterClose());
 
+    this._previousActiveElement = document.activeElement;
     this._boundBlockerClick = this.onBlockerClick.bind(this);
     $('.blocker').on('click', this._boundBlockerClick);
+
+    /**
+     * Trap keyboard focus when modal opens.
+     */
+    const modal = this.$el[0];
+    modal.setAttribute('tabindex', '1');
+    modal.focus();
+
+    const focusableElementsNodeList = modal.querySelectorAll(
+      '#file, #camera, button, input[type="radio"]'
+    );
+    const focusableElements = Array.prototype.slice.call(
+      focusableElementsNodeList
+    );
+
+    this._keyDownListener = this.keyDownListener.bind(null, focusableElements);
+    modal.addEventListener('keydown', this._keyDownListener);
+  },
+
+  keyDownListener(focusableElements, event) {
+    function getNextSelectable(element = focusableElements[0]) {
+      const nextIndex = focusableElements.indexOf(element) + 1;
+      // return first element if the next element doesn't exist
+      return focusableElements[nextIndex]
+        ? focusableElements[nextIndex]
+        : focusableElements[0];
+    }
+
+    function getPreviousSelectable(
+      element = focusableElements[focusableElements.length - 1]
+    ) {
+      const previousIndex = focusableElements.indexOf(element) - 1;
+      // return last element if the previous element doesn't exist
+      return focusableElements[previousIndex]
+        ? focusableElements[previousIndex]
+        : focusableElements[focusableElements.length - 1];
+    }
+
+    //avoid IME composition keydown events
+    //ref: https://developer.mozilla.org/docs/Web/Events/keydown#Notes
+    if (event.isComposing || event.keyCode === 229) return;
+    if (event.keyCode === KeyCodes.TAB) {
+      event.preventDefault();
+      if (event.shiftKey) {
+        getPreviousSelectable(document.activeElement).focus();
+      } else {
+        getNextSelectable(document.activeElement).focus();
+      }
+      // the form submits on 'enter' keypress by default, but if focus is on a radio button,
+      // we want to select it instead
+    } else if (
+      event.keyCode === KeyCodes.ENTER &&
+      document.activeElement.getAttribute('type') === 'radio'
+    ) {
+      event.preventDefault();
```

Comment:
The behavior here seems slightly strange to me when testing it, I suppose I assumed tab would select the radio button. How hard would it be to submit the form if the radio button is already selected, and select if not?